File: writeups/283_Move_Zeroes/ruby/main.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
  count = 0
  nums.filter! do |x|
    if x == 0
      count += 1

      false
    else
      true
    end
  end
  count.times do
    nums.push(0)
  end
end

File: writeups/283_Move_Zeroes/typescript/main.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): number[] {
  const count: number = nums.filter((x) => x === 0).length;
  for (let i = 0; i < count; i++) {
    nums.splice(nums.indexOf(0), 1);
    nums.push(0);
  }

  return nums;
}
